src/PrestaShopBundle/Controller/Admin/CommonController.php

```diff
@@ -358,30 +359,42 @@ public function searchGridAction(
         $gridDefinitionFactoryServiceId,
         $redirectRoute,
         array $redirectQueryParamsToKeep = []
-    ) {
+    )
+    {
+        /** @var ResponseBuilder $responseBuilder */
+        $responseBuilder = $this->get('prestashop.bundle.grid.response_builder');
```

Comment:
It would be better to check for the legacy case first by checking the factory is NOT `FilterableGridDefinitionFactoryInterface` and NOT `AbstractGridDefinitionFactory`
Then you can use `controller_response_builder` and return the response

The idea is just not to get the `prestashop.bundle.grid.response_builder` if it's not needed (therefore not instanciate it and save memory)

Reply:
See https://github.com/PrestaShop/PrestaShop/pull/19506/commits/c47c04caf682a0b64f66d8ed73278ec4641b5185 it should be better now